arrow/array/util.go

```diff
@@ -211,14 +211,67 @@ func RecordFromStructArray(in *Struct, schema *arrow.Schema) arrow.Record {
 // A record batch from JSON is equivalent to reading a struct array in from json and then
 // converting it to a record batch.
 func RecordFromJSON(mem memory.Allocator, schema *arrow.Schema, r io.Reader, opts ...FromJSONOption) (arrow.Record, int64, error) {
-	st := arrow.StructOf(schema.Fields()...)
-	arr, off, err := FromJSON(mem, st, r, opts...)
-	if err != nil {
-		return nil, off, err
+	var cfg fromJSONCfg
```

Comment:
Can we link to the issue here or/and summarize the performance issue so that it's clear why we're doing something rather unusual?

Reply:
Ah, basically we've inlined/specialized `FromJSON` here but with a RecordBuilder?